### ISSUE TITLE
doc-comment validation for Slice operations

### ISSUE THREAD

**Issue by bernardnormier** [zeroc-ice/ice#2998]
As far as I can tell, there is no doc-comment validation in the Slice library, which is rather unexpected.

I think:
 - it should be ok (= no warning/error) to define an operation with no doc-comment at all
 - if you add a doc-comment to an operation:
   - incorrect tag/return/exception should result in an error. For example, a typo in a param name or  an`@param` when there is no parameter
   - `@deprecated` when the operation is not marked deprecated should result in an error
   - missing doc for a param/return/exception should result in a warning

**Comment by bernardnormier** [comment]
```slice
/// I'm not going to document my parameter!
void op(string s);
```

Since there is a doc-comment and your forgot to document `s`, you should get a warning: `missing @param s`.

Now, if you add `@param foo` and there is no foo parameter, you should get an error, not a warning.

>  if a user puts @param myParam with no text after it?

I would not issue a warning or error if there is no text after a `@param xxx`, assuming xxx is an actual parameter of the operation.

**Comment by InsertCreativityHere** [comment]
I don't think we should complain if a user has a doc-comment, but doesn't use every single possible tag in it.
In some languages, it's totally normal not to bother documenting "obvious" parameters, to avoid pointless doc comments that you see all the time in C# and Java:
```
/// This operation sorts a dictionary.
/// @param d: the dictionary to sort.
sortDictionary(d: MyDictionary);
```

Arguably, I think that ommitting the `@param` here would actually _improve_ readability.

----

I agree that we should report inaccuracies though:
1) A user puts `@return` on a void operation.
2) A user put `@param xxx`, but there isn't a corresponding parameter for it.
3) A user put `@throws type` but the types isn't in their exception list.

----

And I'm on the fence about:

> @deprecated when the operation is not marked deprecated should result in an error

But I'm going to open an issue to discuss this, because I think there's a bigger 'problem' here.
